Bottom: b4145069cbef573360cd3564e9c5d14d91026728
Top:    e9205db1ab526e861b393af829cf680914d3f5f5
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-06 16:18:11 +0000

Refactor to use step_id, not int


---

diff --git a/src/hid/common/contour3d.c b/src/hid/common/contour3d.c
index 6aa3189..eb2235e 100644
--- a/src/hid/common/contour3d.c
+++ b/src/hid/common/contour3d.c
@@ -1,6 +1,7 @@
 #include <glib.h>
 #include <stdbool.h>
 
+#include "step_id.h"
 #include "quad.h"
 #include "contour3d.h"
 
diff --git a/src/hid/common/contour3d.h b/src/hid/common/contour3d.h
index 818bd8e..0000492 100644
--- a/src/hid/common/contour3d.h
+++ b/src/hid/common/contour3d.h
@@ -2,7 +2,7 @@ typedef struct {
   edge_ref first_edge;
 
   /* XXX: STEP specific - breaks encapsulation */
-  int face_bound_identifier;
+  step_id face_bound_identifier;
 } contour3d;
 
 contour3d *make_contour3d (edge_ref first_edge);
diff --git a/src/hid/common/edge3d.c b/src/hid/common/edge3d.c
index 263f986..35a8846 100644
--- a/src/hid/common/edge3d.c
+++ b/src/hid/common/edge3d.c
@@ -2,6 +2,7 @@
 #include <stdlib.h>
 #include <glib.h>
 
+#include "step_id.h"
 #include "edge3d.h"
 
 edge_info *
diff --git a/src/hid/common/edge3d.h b/src/hid/common/edge3d.h
index f54468c..3080865 100644
--- a/src/hid/common/edge3d.h
+++ b/src/hid/common/edge3d.h
@@ -14,8 +14,8 @@ typedef struct
   double radius;
 
   /* XXX: STEP specific - breaks encapsulation */
-  int infinite_line_identifier;
-  int edge_identifier;
+  step_id infinite_line_identifier;
+  step_id edge_identifier;
 } edge_info;
 
 edge_info *make_edge_info (void);
diff --git a/src/hid/common/face3d.c b/src/hid/common/face3d.c
index 0926001..5f1df0a 100644
--- a/src/hid/common/face3d.c
+++ b/src/hid/common/face3d.c
@@ -1,6 +1,7 @@
 #include <glib.h>
 #include <stdbool.h>
 
+#include "step_id.h"
 #include "quad.h"
 #include "contour3d.h"
 #include "appearance.h"
diff --git a/src/hid/common/face3d.h b/src/hid/common/face3d.h
index c630c17..1ca9eb4 100644
--- a/src/hid/common/face3d.h
+++ b/src/hid/common/face3d.h
@@ -12,9 +12,9 @@ typedef struct {
   appearance *appear;
 
   /* XXX: STEP specific - breaks encapsulation */
-  int surface_identifier;
-  int face_identifier;
-  int face_bound_identifier;
+  step_id surface_identifier;
+  step_id face_identifier;
+  step_id face_bound_identifier;
 } face3d;
 
 face3d *make_face3d (void);
diff --git a/src/hid/common/vertex3d.c b/src/hid/common/vertex3d.c
index 1704b2b..74c286c 100644
--- a/src/hid/common/vertex3d.c
+++ b/src/hid/common/vertex3d.c
@@ -1,5 +1,7 @@
+#include <glib.h>
 #include <stdlib.h>
 
+#include "step_id.h"
 #include "vertex3d.h"
 
 static int global_vertex3d_count;
diff --git a/src/hid/common/vertex3d.h b/src/hid/common/vertex3d.h
index 3625c5e..8bb6ec9 100644
--- a/src/hid/common/vertex3d.h
+++ b/src/hid/common/vertex3d.h
@@ -6,7 +6,7 @@ typedef struct
   int id;
 
   /* XXX: STEP specific - breaks encapsulation */
-  int vertex_identifier;
+  step_id vertex_identifier;
 } vertex3d;
 
 vertex3d *make_vertex3d (double x, double y, double z);
